File: lista de exercicios/questão 14/questao.py
class Configuracao:
    _instancia = None

    def __new__(cls, *args, **kwargs):
        if cls._instancia is None:
            cls._instancia = super().__new__(cls, *args, **kwargs)
        return cls._instancia

# Exemplo de uso
config1 = Configuracao()
config2 = Configuracao()

print(config1 is config2)  # Saída: True (mesma instância)

File: lista de exercicios/questão 14/questao.go
package main

import (
    "fmt"
    "sync"
)

// Struct para configuração
type Configuracao struct{}

var instancia *Configuracao
var once sync.Once

// Função para obter a instância única
func GetInstancia() *Configuracao {
    once.Do(func() {
        instancia = &Configuracao{}
    })
    return instancia
}

func main() {
    config1 := GetInstancia()
    config2 := GetInstancia()

    fmt.Println(config1 == config2)  // Saída: true (mesma instância)
}
